### ISSUE TITLE
[routing] queryroutes returns "insufficient local balance" due to bandwidth hints error.

### ISSUE THREAD

**Issue by RostyslavAntonyshyn** [lightningnetwork/lnd#4232]
### Background

I am facing the issue ( might be related to #4181 ) when after some manipulations queryroutes returns "insufficient local balance" due to bandwidth hints error.

### Your environment

* version of `lnd` - 0.10.0 - rc5
* which operating system - ubuntu
* Using litecoin on mainnet

### Steps to reproduce

No exact way to reproduce this ( was running a lot of queryroutes one by one, hodl invoices, quickly sending multiple amt of payments etc. )

### Actual behavior

Actually lnd returns error on queryroutes, and what is most confusing, is that once this happened it happens always. Only closing channel and creating new one resolves this.

From lnd logs:

```
2020-04-29 17:49:05.248 [DBG] CRTR: Searching for path to 026776d2e93d2509c1bb7e94b158fd1a676ac6c97e4c0a8e01dcfdef4dcf42dac5, sending 1000000 mSAT
2020-04-29 17:49:05.249 [ERR] LNWL: ChannelPoint(a7a212ec54a410e40b6200e982452ed0f87d64dffd65974616565bc935ea6d75:0): Unable to fetch available balance: parent entry 20 for update 20 had zero local add height
2020-04-29 17:49:05.249 [DBG] CRTR: Pathfinding perf metrics: nodes=0, edges=0, time=418.73µs
2020-04-29 17:49:05.250 [ERR] RPCS: [/lnrpc.Lightning/QueryRoutes]: insufficient local balance
2020-04-29 17:49:17.362 [INF] DISC: Retransmitting 0 outgoing channels and our refreshed node announcement
```

My channels:

```
{
    "channels": [
        {
            "active": true,
            "remote_pubkey": "0375e7d882b442785aa697d57c3ed3aef523eb2743193389bd205f9ae0c609e6f3",
            "channel_point": "8cb64db0d834c52745d4c15fa9e5d23b96a2329c18fa9b29f36eb01afa0e2858:0",
            "chan_id": "2014604369251074048",
            "capacity": "275000",
            "local_balance": "90",
            "remote_balance": "274722",
            "commit_fee": "278",
            "commit_weight": "552",
            "fee_per_kw": "260",
            "unsettled_balance": "0",
            "total_satoshis_sent": "0",
            "total_satoshis_received": "90",
            "num_updates": "16",
            "pending_htlcs": [
            ],
            "csv_delay": 576,
            "private": false,
            "initiator": false,
            "chan_status_flags": "ChanStatusDefault",
            "local_chan_reserve_sat": "54600",
            "remote_chan_reserve_sat": "54600",
            "static_remote_key": true,
            "commitment_type": "STATIC_REMOTE_KEY",
            "lifetime": "4862",
            "uptime": "4856",
            "close_address": "",
            "push_amount_sat": "0",
            "thaw_height": 0
        },
        {
            "active": true,
            "remote_pubkey": "0375e7d882b442785aa697d57c3ed3aef523eb2743193389bd205f9ae0c609e6f3",
            "channel_point": "a7a212ec54a410e40b6200e982452ed0f87d64dffd65974616565bc935ea6d75:0",
            "chan_id": "2014604369251139584",
            "capacity": "275000",
            "local_balance": "274766",
            "remote_balance": "51",
            "commit_fee": "234",
            "commit_weight": "600",
            "fee_per_kw": "253",
            "unsettled_balance": "0",
            "total_satoshis_sent": "51",
            "total_satoshis_received": "0",
            "num_updates": "42",
            "pending_htlcs": [
            ],
            "csv_delay": 576,
            "private": false,
            "initiator": true,
            "chan_status_flags": "ChanStatusDefault",
            "local_chan_reserve_sat": "54600",
            "remote_chan_reserve_sat": "54600",
            "static_remote_key": true,
            "commitment_type": "STATIC_REMOTE_KEY",
            "lifetime": "4862",
            "uptime": "4856",
            "close_address": "",
            "push_amount_sat": "0",
            "thaw_height": 0
        }
    ]
}
```

**Comment by Roasbeef** [comment]
Which version of `lnd` did you update from?

**Comment by RostyslavAntonyshyn** [comment]
> Which version of `lnd` did you update from?

lnd-0.8.2

**Comment by joostjager** [comment]
A error similar to `Unable to fetch available balance: parent entry 20 for update 20 had zero local add height` is reproduced in https://github.com/lightningnetwork/lnd/pull/4193:

Something isn't right in the channel state machine. Could be a still lingering channel re-establish issue.

**Comment by joostjager** [comment]
@Crypt-iQ do you think this is addressed with the 0.11 release?

**Comment by Crypt-iQ** [comment]
@joostjager This shouldn't be happening anymore with 0.11
